### PR TITLE
*: use another etcd client for election

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Ref #6403.

### What is changed and how does it work?
This PR tries to use another etcd client for the election to prevent the throttling of one connection.

```commit-message
```

### Check List

Tests

- Manual test (TODO)

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/utils/etcdutil/etcdutil.go [hunk 1]
```diff
@@ -193,24 +193,18 @@ func EtcdKVPutWithTTL(ctx context.Context, c *clientv3.Client, key string, value
...
-	if err != nil {
-		return nil, nil, errs.ErrNewEtcdClient.Wrap(err).GenWithStackByCause()
-	}
-	httpClient := createHTTPClient(tlsConfig)
-	return client, httpClient, nil
-}
-
 // CreateClients creates etcd v3 client and http client.
-func CreateClients(tlsConfig *tls.Config, acUrls url.URL) (*clientv3.Client, *http.Client, error) {
+func CreateClients(tlsConfig *tls.Config, acUrls url.URL) (*clientv3.Client, *clientv3.Client, *http.Client, error) {
```
Comment (lhy1024) [inline pkg/utils/etcdutil/etcdutil.go:197]: return two etcdClient? maybe we can directly create electionClient with another function?

File: server/server.go [hunk 2]
```diff
@@ -356,18 +358,18 @@ func (s *Server) startEtcd(ctx context.Context) error {
 	failpoint.Inject("memberNil", func() {
 		time.Sleep(1500 * time.Millisecond)
 	})
-	s.member = member.NewMember(etcd, s.client, etcdServerID)
+	s.member = member.NewMember(etcd, s.electionClient, etcdServerID)
 	return nil
 }
 
-func startClient(cfg *config.Config) (*clientv3.Client, *http.Client, error) {
+func startClient(cfg *config.Config) (*clientv3.Client, *clientv3.Client, *http.Client, error) {
```
Comment (bufferflies) [inline server/server.go:365]: why not use array?

File: pkg/utils/etcdutil/etcdutil.go (hunk 1 above)
Comment (binshi-bing) [reply pkg/utils/etcdutil/etcdutil.go:197]: +1. The change in pr isn't clean

Comment (binshi-bing) [reply pkg/utils/etcdutil/etcdutil.go:197]: Just call "electionClient, err := createEtcdClient(tlsConfig, acUrls)" to create a dedicate election etcd client is clearer.

**Review by binshi-bing — APPROVED** [review]

File: pkg/utils/etcdutil/etcdutil.go (hunk 1 above)
Comment (rleungx) [reply pkg/utils/etcdutil/etcdutil.go:197]: Hmmm... I'm trying to avoid duplicated config initialization.

Comment (lhy1024) [reply pkg/utils/etcdutil/etcdutil.go:197]: just to add a `createEtcdClient` in server.go?

Comment (nolouch) [reply pkg/utils/etcdutil/etcdutil.go:197]: +1, we can create two clients in `startClient` not here.

**Review by lhy1024 — APPROVED** [review]

**Review by nolouch — APPROVED** [review]
lgtm

**Pull request merged**